python/string/stringFollowPattern.py

```diff
@@ -0,0 +1,34 @@
+# Check if the string follows a pattern
+
+"""
+Given a string and a pattern, does the string follow the given pattern
+eg - String - "red black red"
+    Pattern - "aba"
+
+    result = True
+"""
+
+
+def wordPattern(str, pattern):
+    if len(pattern) == 0 or len(str) == 0: return False
+
+    word_list = str.split()
+    pattern_list = list(pattern)
+    if len(word_list) != len(pattern_list): return False
+
+    w2p = {}
+    p2w = {}
```

Comment:
Only one dictionary, `pattern_to_word` should be sufficient. the other is not required here.